scripts/prepdocslib/integratedvectorizerstrategy.py

```diff
@@ -0,0 +1,207 @@
+            search_images=False,
+        )
+        await search_manager.create_index(
+            [
+                AzureOpenAIVectorizer(
+                    name="myOpenAI",
+                    kind="azureOpenAI",
+                    azure_open_ai_parameters=AzureOpenAIParameters(
+                        resource_uri=f"https://{self.embeddings.open_ai_service}.openai.azure.com",
+                        deployment_id=self.embeddings.open_ai_deployment,
+                        auth_identity=SearchIndexerDataUserAssignedIdentity(
+                            user_assigned_identity=f"/subscriptions/{self.subscriptionId}/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{self.search_user_assigned_identity}"
+                        ),
+                    ),
+                ),
+            ]
+        )
+
+        # create indexer client
+        ds_client = search_info.create_search_indexer_client()
+        ds_container = SearchIndexerDataContainer(name=self.blob_manager.container)
+        print(await self.blob_manager.get_blob_sas_365days())
+        data_source_connection = SearchIndexerDataSourceConnection(
+            name=f"{search_info.index_name}-blob",
+            type="azureblob",
+            connection_string=await self.blob_manager.get_blob_sas_365days(),
+            container=ds_container,
+        )
+        data_source = await ds_client.create_or_update_data_source_connection(data_source_connection)
+
+        embedding_skillset = await self.createEmbeddingSkill(search_info.index_name)
+        await ds_client.create_or_update_skillset(embedding_skillset)
+
+    async def run(self, search_info: SearchInfo):
+        if self.document_action == DocumentAction.Add:
+            files = self.list_file_strategy.list()
+            async for file in files:
+                try:
+                    blob_sas_uris = await self.blob_manager.upload_blob(file)
+                finally:
+                    if file:
+                        file.close()
+
+        # Create an indexer
+        indexer_name = f"{search_info.index_name}-indexer"
+
+        indexer = SearchIndexer(
+            name=indexer_name,
+            description="Indexer to index documents and generate embeddings",
+            skillset_name=f"{search_info.index_name}-skillset",
+            target_index_name=search_info.index_name,
+            data_source_name=f"{search_info.index_name}-blob",
+            # Map the metadata_storage_name field to the title field in the index to display the PDF title in the search results
+            field_mappings=[FieldMapping(source_field_name="metadata_storage_name", target_field_name="title")],
+        )
+
+        indexer_client = search_info.create_search_indexer_client()
+
+        indexer_result = await indexer_client.create_or_update_indexer(indexer)
+
+        print(indexer_result)
+
+        # Run the indexer
+        await indexer_client.run_indexer(indexer_name)
```

Comment:
to maintain parity with the existing file strategy - would you consider polling until the indexer finishes running? The user can cancel the script if they don't want to wait

Reply:
I would like to keep it simple. Its better to view the status in azure portal.